### ISSUE TITLE
Add option to serve an https endpoint instead of http

### ISSUE THREAD

**Issue by halberom** [goss-org/goss#966]
**Describe the feature:**
Some companies/solutions require services to all use TLS/HTTPS.  At present goss only serves HTTP.

**Describe the solution you'd like**
Add ability to serve an https endpoint given a certificate.

E.g.
```
goss --serve-https --cert-key /path/to/cert.key --cert /path/to/cert.crt
```

**Describe alternatives you've considered**
NA

**Comment by aelsabbahy** [comment]
Makes sense, marking this as approved. I haven't looked too close to Go's `net/http` to see how well it's supported in the stdlib. I would be against pulling in a third party https service into the package though.
